Application/EdFi.Ods.AdminApi/Infrastructure/Security/SecurityExtensions.cs

```diff
@@ -80,12 +77,14 @@ public static class SecurityExtensions
             opt.DefaultChallengeScheme = JwtBearerDefaults.AuthenticationScheme;
         }).AddJwtBearer(opt =>
         {
-            opt.Authority = authority;
+            opt.Authority = issuer;
```

Comment:
In my tests we get an `invalid token` error without setting `Authority` here.

Reply:
Apparently I overlooked that line when I was looking for "authority" (probably had case sensitive search on). I had also been searching the web for what this "Authority" is and couldn't find documentation. Was driving me crazy. But it does point out another line: 102 sets `ValidIssuer = authority`. Before changing anything else, I want to find the documentation on this.

Reply:
> `ValidIssuer`: Gets or sets a String that represents a valid issuer that will be used to check against the token's issuer. The default is null.

https://learn.microsoft.com/en-us/dotnet/api/microsoft.identitymodel.tokens.tokenvalidationparameters?view=msal-web-dotnet-latest

> Authority is the address of the token-issuing authentication server. The JWT bearer authentication middleware will use this URI to find and retrieve the public key that can be used to validate the token’s signature. It will also confirm that the iss parameter in the token matches this URI.

https://devblogs.microsoft.com/dotnet/jwt-validation-and-authorization-in-asp-net-core/

So, the Authority "confirms" `iss` in the JWT. And `iss` is "issuer". Same article mentions `ValidIssuer` without describing, and shows the values being the same. So far, it is sounding as if these are redundant properties.

https://auth0.com/blog/how-to-validate-jwt-dotnet/ - similarly does not distinguish between "authority" and "issuer".

But, this last article shows `ValidIssuer` being a domain, not a URI. And it is true that the data type is a `string`, whereas the data type for the OpenIdDict `Issuer` is a `URI`.

I've also looked through https://github.com/openiddict/openiddict-samples, which sadly doesn't have a JWT example. And Microsoft's documentation of `AddJwtBearer` does not give any example values.

... so here's the conclusion I'm reaching, hopefully you see the same thing (please do confirm that I'm not off my rocker again):

* For OpenIdDict configuration (via `.AddServer`), the Issuer is a URI to the OAuth provider
* For ASP.NET JWT bearer token configuration,
  * The Authority is the same as the Issuer above
  * The ValidIssuer should be the _domain_ only from the Issuer

If that is the case, it might make sense to have only one appSetting: the URI for the Issuer. Then we just extract the `.Host` value from that URI when setting the `ValidIssuer`.

Reply:
Setting `ValidIssuer` to domain only failed. I've reverted it back to matching the `Issuer` and `Authority`

Reply:
Fascinating. What horrible documentation.